AppController/lib/djinn_job_data.rb

```diff
@@ -117,6 +117,22 @@ def method_missing(id, *args, &block)
     super
   end
 
+  # In the process of removing roles, db_slave is no longer added to non
+  # db_master nodes. A node that is a database and is not the db_master is
+  # now considered a db_slave.
+  def is_db_slave?
+    return @jobs.include?('db_slave') ||
```

Comment:
I kept the checks for the old roles to cover the upgrade case, but I'm not sure how necessary it is. Opinions?